pkg/operator/staticpod/prune/cmd.go

```diff
@@ -0,0 +1,116 @@
+package prune
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type PruneOptions struct {
+	MaxEligibleRevisionID int
+	ProtectedRevisionIDs  []int
+
+	ResourceDir   string
+	StaticPodName string
+}
+
+func NewPruneOptions() *PruneOptions {
+	return &PruneOptions{}
+}
+
+func NewPrune() *cobra.Command {
+	o := NewPruneOptions()
+
+	cmd := &cobra.Command{
+		Use:   "prune",
+		Short: "Prune static pod installer revisions",
+		Run: func(cmd *cobra.Command, args []string) {
+			glog.V(1).Info(cmd.Flags())
+			glog.V(1).Info(spew.Sdump(o))
+
+			if err := o.Validate(); err != nil {
+				glog.Fatal(err)
+			}
+			if err := o.Run(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	o.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (o *PruneOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.IntVar(&o.MaxEligibleRevisionID, "max-eligible-id", o.MaxEligibleRevisionID, "highest revision ID to be eligible for pruning")
+	fs.IntSliceVar(&o.ProtectedRevisionIDs, "protected-ids", o.ProtectedRevisionIDs, "list of revision IDs to preserve (not delete)")
+	fs.StringVar(&o.ResourceDir, "resource-dir", o.ResourceDir, "directory for all files supporting the static pod manifest")
+	fs.StringVar(&o.StaticPodName, "static-pod-name", o.StaticPodName, "name of the static pod")
+}
+
+func (o *PruneOptions) Validate() error {
+	if len(o.ResourceDir) == 0 {
+		return fmt.Errorf("--resource-dir is required")
+	}
+	if o.MaxEligibleRevisionID == 0 {
+		return fmt.Errorf("--max-eligible-id is required")
+	}
+	if len(o.StaticPodName) == 0 {
+		return fmt.Errorf("--static-pod-name is required")
+	}
+
+	return nil
+}
+
+func (o *PruneOptions) Run() error {
+	protectedIDs := sets.NewInt(o.ProtectedRevisionIDs...)
+
+	files, err := ioutil.ReadDir(o.ResourceDir)
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		// If the file is not a resource directory...
+		if !file.IsDir() {
+			continue
+		}
+		// And doesn't match our static pod prefix...
+		if !strings.HasPrefix(file.Name(), o.StaticPodName) {
+			continue
+		}
+
+		// Split file name to get just the integer revision ID
+		fileSplit := strings.Split(file.Name(), fmt.Sprintf("%s-", o.StaticPodName))
```

Comment:
seems like ensuring the remainder after the index of the last `-` is an integer will be more accurate.  Ones that aren't ints are not failures, they are just like `mypod-isspecial-1` when you wanted `mypod-1`

Reply:
> seems like ensuring the remainder after the index of the last `-` is an integer will be more accurate. Ones that aren't ints are not failures, they are just like `mypod-isspecial-1` when you wanted `mypod-1`

I'll take a followup

Reply:
> I'll take a followup

along with a test that ensure it doesn't slip.  prefix checking is the bane of testing.